admin_ui/backend/tests/test_updates_api.py

```
@@ -0,0 +1,131 @@
+        ("feature/foo", None),
+    ],
+)
+def test_updater_pull_preference_only_for_release_targets(ref: str, expected: str | None) -> None:
+    assert system._updater_prefer_pull_ref_for_update_target(ref) == expected
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        "agent",
+        "/tmp/agent;rm",
```

Ruff (0.15.18)

[error] Probable insecure usage of temporary file or directory: "/tmp/agent;rm"

(S108)